File: Backend/Ecommerce/src/main/java/com/example/Ecommerce/product/dto/Product.java
package com.example.Ecommerce.product.dto;

import com.example.Ecommerce.Enum.Status;
import com.example.Ecommerce.audit.bean.Audit;
import lombok.Data;


@Data
public class Product extends Audit {

    private Integer id;
    private String name;
    private String description;
    private String imageUrl;
    private Status status;
    private Integer price;
    private ProductCategory productCategory;
}

File: Frontend/src/app/components/dashboard/models/Product.ts
import { ProductCategory } from "./ProductCategory";

export interface Product {
    id?:Number;
    name?:Number;
    description?:string;
    price?:Number;
    imageUrl?:string;
    status?:string;
    createdAt?:string;
    modifiedAt?:string;
    productCategory?:ProductCategory;
}